### DISCUSSION TITLE
Azure a segregation of duties

### DISCUSSION THREAD

**Discussion by maciej-skorupka** [AzureCommunityPL/group#286] [Pomoc]
Cześć.

Nie wiem jak to wygląda w Waszych organizacjach, ale w niektórych jest wymóg segregation of duties i ogólnego podziału kompetencji. Coś co przy wersji on premise generowało zespoły infrastruktury z linuxowcami, windowsowcami, sieciowcami,  dyskowcami a gdzieś obok tego zespoły aplikacji, cicd aplikacyjnego itp.

Możliwe, że to naiwna myśl, ale w chmurze to po części wydaje się być do ogarnięcia jednym pipelinem, który szedłby od dołu do góry konkretnego rozwiazania. Czyli stawiałby vnety, subnety, nsg firewalle itd i na tym już storage, klaster, adf czy czego tam trzeba.

Tak robicie? Czy raczej dalej podobne podziały infrastruktura-aplikacje istnieją? Macie wtedy coś w stylu oddzielnych pipelinow, które budują całą infrastrukturę a potem oddzielnych dla aplikacji? Dzielicie takie pipeliny infrastruktury na mniejsze pipeliny na przykład oddzielny dla VMek, oddzielny dla sieci, oddzielny dla storage itp?

Wiem, że to dość ogólne pytanie i pewnie różne odpowiedzi mogą być w zależności od firmy i rozwiązania. Niemniej jednak jestem ciekaw doświadczeń.

**Comment by TomaszOledzki** [comment]
Pull Requesty do brancza z którego robisz paczkę releasową i same releasey ktoś musi akceptować. Możesz nadal mieć podział kompetencji i użyć polisy na approvalach w Azure DevOps. Ładnie do tego zapinasz testy i masz piękny pipeline.

**Reply by maciej-skorupka** [reply]
Czyli rozumiem, że proponujesz pisanie pipelinu stawiającego środowisko od sieci po zasoby aplikacji w jednym. I segregation of duties na poziomie zatwierdzania przez inne osoby wykonania pipelinu.

**Comment by piotrgwiazda** [comment]
Zwykle mam ze 2-3 warstwy Terraforma od landing zona do aplikacji. Oprócz
tego kunernetes nieźle oddziela infrę od app.
Pamiętaj że ze 2 warstwy segregation of duties załatwia provider.

Zobacz (autoremlama)
https://blog.gft.com/pl/2020/03/04/secure-terraform-delivery-pipeline-best-practices-part-1-2/

Oraz
https://github.com/Azure/caf-terraform-landingzones



niedz., 30 maj 2021, 13:42 użytkownik maciej-skorupka <
***@***.***> napisał:
